skbuild/resources/cmake/UseF2PY.cmake

```diff
@@ -0,0 +1,149 @@
+#   # Note: In this case, either one of these arguments may be omitted; their
+#   # value would have been inferred from that of the other.
+#   add_f2py_target(f2py_code f2py_code.pyf)
+#
+#   add_library(f2py_code MODULE ${f2py_code})
+#   target_link_libraries(f2py_code ...)
+#
+#=============================================================================
+# Copyright 2011 Kitware, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+find_package(F2PY REQUIRED)
+
+get_property(languages GLOBAL PROPERTY ENABLED_LANGUAGES)
+
+function(add_f2py_target _name)
+  set(options PY2 PY3)
+  set(options1 OUTPUT_VAR)
+  cmake_parse_arguments(_args "${options}" "${options1}" "" ${ARGN})
+
+  list(GET _args_UNPARSED_ARGUMENTS 0 _arg0)
+
+  # if provided, use _arg0 as the input file path
+  if(_arg0)
+    set(_source_file ${_arg0})
+
+  # otherwise, must determine source file from name, or vice versa
+  else()
+    get_filename_component(_name_ext "${_name}" EXT)
+
+    # if extension provided, _name is the source file
+    if(_name_ext)
+      set(_source_file ${_name})
+      get_filename_component(_name "${_source_file}" NAME_WE)
+
+    # otherwise, assume the source file is ${_name}.pyf
+    else()
+      set(_source_file ${_name}.pyf)
+    endif()
+  endif()
+
+  set(_embed_main FALSE)
+
+  if("C" IN_LIST languages)
+    set(_output_syntax "C")
+  else()
+    message(FATAL_ERROR "C must be enabled to use F2PY")
+  endif()
+
+  set(extension "c")
+
+  set(generated_file "${CMAKE_CURRENT_BINARY_DIR}/${_name}.${extension}")
+  set_source_files_properties(${generated_file} PROPERTIES GENERATED TRUE)
+
+  set(_output_var ${_name})
+  if(_args_OUTPUT_VAR)
+      set(_output_var ${_args_OUTPUT_VAR})
+  endif()
+  set(${_output_var} ${generated_file} PARENT_SCOPE)
+
+  file(RELATIVE_PATH generated_file_relative
+      ${CMAKE_BINARY_DIR} ${generated_file})
+
+  set(comment "Generating ${_output_syntax} source ${generated_file_relative}")
+
+  # Get the include directories.
+  get_source_file_property(pyf_location ${_source_file} LOCATION)
+  get_filename_component(pyf_path ${pyf_location} PATH)
+  get_directory_property(cmake_include_directories
+                         DIRECTORY ${pyf_path}
+                         INCLUDE_DIRECTORIES)
+  list(APPEND cython_include_directories ${cmake_include_directories})
+
+  # Include directory arguments.
+  set(include_directory_arg "")
```

Comment:
I think the setting of `include_directory_arg` can be removed. It doesn't seem to be used.